src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -77,26 +76,34 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
 
             try
             {
-                // Get destination type from secret store.
-                DestinationInfo destinationInfo = await GetDestinationInfo(cancellationToken);
+                // Get destination type from secret store and connect to the destination using appropriate client.
+                await GetDestinationInfoAndConnect(cancellationToken);
 
-                // Connect to the destination using appropriate client.
-                _exportDestinationClient = _exportDestinationClientFactory.Create(destinationInfo.DestinationType);
-
-                await _exportDestinationClient.ConnectAsync(destinationInfo.DestinationConnectionString, cancellationToken, _exportJobRecord.Id);
+                // If we are resuming a job, we can detect that by checking the progress info from the job record.
+                // If it is null, then we know we are processing a new job.
+                if (_exportJobRecord.Progress != null)
+                {
+                    // Update our internal mapping with list of files that we have already committed.
```

Comment:
isn't the _exportJobRecord.Output and _resourceTypeToFileInforMapping the same thing?

Reply:
Ya. Initially the Output field was a List and I had written the code based on that. But then it made more sense to change it into a Dictionary but I missed out on updating the code to take advantage of the fact that both Output and _resourceTypeToFileInforMapping are using the same datastructure. I have updated the code to remove the copying over part.